internal/proxy.go

```diff
@@ -11,23 +11,31 @@ import (
 )
 
 type ProxyConnectionHandler struct {
-	// keep this as first element of struct to guarantee 64-bit alignmen ton 32-bit machines.
+	// keep this as first element of struct to guarantee 64-bit alignment ton 32-bit machines.
 	// atomic.* functions crash if the operand is not 64-bit aligned.
 	// See https://github.com/golang/go/issues/599
-	failures    int64
-	address     string
-	flushTicker *time.Ticker
-	done        chan struct{}
-	mtx         sync.RWMutex
-	conn        net.Conn
-	writer      *bufio.Writer
+	failures         int64
+	address          string
+	flushTicker      *time.Ticker
+	done             chan struct{}
+	mtx              sync.RWMutex
+	conn             net.Conn
+	writer           *bufio.Writer
+	internalRegistry *MetricRegistry
+
+	writeSuccesses *DeltaCounter
+	writeErrors    *DeltaCounter
 }
 
-func NewProxyConnectionHandler(address string, flushInterval time.Duration) ConnectionHandler {
-	return &ProxyConnectionHandler{
-		address:     address,
-		flushTicker: time.NewTicker(flushInterval),
+func NewProxyConnectionHandler(address string, flushInterval time.Duration, prefix string, internalRegistry *MetricRegistry) ConnectionHandler {
```

Comment:
Couple of points here:
* This is an externally facing constructor. Changing the signature will break backwards compat.
* But also the internal registry should be an internal thing (users should not need to pass it in?). We should default to a prefix and create the internal registry, but allow users to disable this (as described here: https://github.com/wavefrontHQ/wavefront-sdk-go/issues/35)

Reply:
Nevermind, ignore my previous comment. It's probably okay since this is internal.